pkg/agent/endpoints/workload/handler.go

```diff
@@ -130,6 +136,11 @@ func (h *Handler) FetchJWTBundles(req *workload.JWTBundlesRequest, stream worklo
 					telemetry.Seconds: time.Since(start).Seconds,
 					telemetry.PID:     pid,
 				}).Warn("Took >1 second to send JWT bundle to PID")
+			} else {
+				h.Log.WithFields(logrus.Fields{
```

Comment:
with my comment above about "picking up fields as we go", be careful of pitfalls like in this situation; here we would **not** want to keep adding TTL and PID fields for eternity :)